test/Dialect/FIRRTL/lower-signatures.mlir

```diff
@@ -49,3 +49,39 @@ firrtl.circuit "PreserveLocation" {
 // CHECK-LOC: [[LOC]] = loc("someLoc":9001:1)
 #moduleLoc = loc("wrongLoc":42:1)
 #instLoc = loc("someLoc":9001:1)
+
+// Internal paths should be expanded
+firrtl.circuit "InternalPaths"  {
+  // CHECK: firrtl.extmodule private @BlackBox
+  // CHECK-SAME: out bundle_a: !firrtl.uint<32>, out bundle_b: !firrtl.uint<23>
+  // CHECK-SAME: out array_0: !firrtl.uint<1>, out array_1: !firrtl.uint<1>
+  // CHECK-SAME: out probe: !firrtl.probe<uint<32>>
+  // CHECK-SAME: internalPaths = [
+  // CHECK-SAME: #firrtl.internalpath
+  // CHECK-SAME: #firrtl.internalpath
+  // CHECK-SAME: #firrtl.internalpath
+  // CHECK-SAME: #firrtl.internalpath
+  // CHECK-SAME: #firrtl.internalpath<"some_probe">
```

Comment:
This test doesn't cover all the logic being implemented.  Expecially the .[fieldname] additions